editor/plugins/script_text_editor.cpp

```diff
@@ -605,6 +605,48 @@ void ScriptTextEditor::_update_color_constructor_options() {
 	}
 }
 
+void ScriptTextEditor::_update_background_text() {
+	// Clear background lines.
+	CodeEdit *te = code_editor->get_text_editor();
+	for (int i = 0; i < te->get_line_count(); i++) {
+		bool is_folded_code_region = te->is_line_code_region_start(i) && te->is_line_folded(i);
+		te->set_line_background_color(i, is_folded_code_region ? folded_code_region_color : Color(0, 0, 0, 0));
+	}
+
+	// Set the warning background.
+	if (warning_line_color.a != 0.0) {
+		for (const ScriptLanguage::Warning &W : warnings) {
+			int folder_line_header = te->get_folded_line_header(W.end_line - 2);
+			bool is_folded = folder_line_header != (W.end_line - 2);
+
+			if (is_folded) {
+				te->set_line_background_color(folder_line_header, warning_line_color);
+			} else if (W.end_line - W.start_line > 0 && W.end_line - W.start_line < 20) {
+				// If the warning spans below 20 lines (arbitrary), set the background color for all lines.
+				// (W.end_line - W.start_line > 0) ensures that we set the background for single line warnings.
+				for (int i = W.start_line - 1; i < W.end_line - 1; i++) {
+					te->set_line_background_color(i, warning_line_color);
+				}
+			} else {
+				// Otherwise, just set the background color for the start line of the warning.
+				te->set_line_background_color(W.start_line - 1, warning_line_color);
+			}
+		}
+	}
+
+	// Set the error background.
+	for (const ScriptLanguage::ScriptError &E : errors) {
```

Comment:
```suggestion
	if (marked_line_color.a != 0.0) {
		for (const ScriptLanguage::ScriptError &error : errors) {
```
This should check for alpha, and `error` is more clear.